### ISSUE TITLE
include libigl in ExternalLibs (which includes Eigen lib instead of itkeigen)

### ISSUE THREAD

**Issue by sheryjoe** [SCIInstitute/ShapeWorks#93]
- Transforms and SurfaceReconstructions are now using itkeigen
- All instances of Eigen lib should be removed
- Replace #include "../Eigen/XX" with #include <itkeigen/Eigen/XX>

**Comment by cchriste** [comment]
This is a continuing story of how best to include Eigen [v3.3.7].
The current best answer (imo) is to include `libigl`, and the best way to do that might be to simply clone it into ExternalLibs since cmake's ExternalProject_Add couldn't quite do the job (though it's been updated since we tried, so perhaps [its successor](https://cmake.org/cmake/help/latest/module/ExternalProject.html) will work), but it's complicated and time-consuming and so I'm proposing a straightforward solution that will work today and that we simply create a maintenance reminder every few months to update.

The most important thing is to explicitly specify Eigen 3.3.7 because for some reason libigl (last time I looked a couple months back) currently includes an older version by default, but it works fine with the latest.

I added Karthik since he's using functionality duplicated from libigl, and the library's version is built for easy parallelization and therefore more efficient and less effort to maintain, and I added Alan since he has a lot of expertise to contribute to adding an external library.

It's even possible that a conda version of libigl is a now available.
